src/main/scala/java/util/logging/Logger.scala

```diff
@@ -0,0 +1,322 @@
+
+object Logger {
+
+  val GLOBAL_LOGGER_NAME: String =  "global"
+
+  // Not implemented, deprecated on JDK 1.8
+  //val global: Logger
+
+  private val defaultLogLevel: Level = Level.ALL
+
+  private val loggers: mutable.Map[String, Logger] = mutable.Map.empty
+
+  // Root is not visible to the outside but gives defaults
+  private[this] val rootLogger: Logger = {
+    val l = new Logger("", null)
+    l.setLevel(defaultLogLevel)
+    l.setUseParentHandlers(false)
+    l.setParent(null)
+    l
+  }
+
+  private[this] val globalLogger: Logger = {
+    val l = new Logger(GLOBAL_LOGGER_NAME, null)
+    l.setLevel(defaultLogLevel)
+    l.setUseParentHandlers(true)
+    l.setParent(rootLogger)
+    l
+  }
+
+  def getGlobal(): Logger = globalLogger
+
+  def getLogger(name: String): Logger = {
+    if (name == null)
+      throw new NullPointerException("Logger name cannot be null")
+
+    loggers.getOrElseUpdate(name, {
+      val l = new Logger(name, null)
+      l.setLevel(null)
+      l.setUseParentHandlers(true)
+      l.setParent(rootLogger)
+      l
+    })
+  }
+
+  // Not implemented, no resource bundle in scala.js
+  //def getLogger(name: String, resourceBundle: String): Logger
+
+  private[logging] def findParent(logger: Logger): Option[Logger] = {
+    @tailrec
+    def go(s: List[String]): Option[Logger] = s match {
+      case Nil => None
+
+      case b if loggers.contains(b.mkString(".")) =>
+        loggers.get(b.mkString("."))
+
+      case b => go(b.dropRight(1))
+    }
+
+    go(Option(logger.getName).getOrElse("").split("\\.").toList.dropRight(1))
+  }
+
+  def getAnonymousLogger(): Logger = {
+    // No references to anonymous loggers are kept
+    val l = new Logger(null, null)
+    l.setLevel(null)
+    l.setUseParentHandlers(true)
+    l.setParent(rootLogger)
+    l
+  }
+
+  // Not implemented, no resource bundle in scala.js
+  //def getAnonymousLogger(resourceBundle: String):Logger
+}
+
+class Logger(name: String, resourceBundle: String) {
+
+  private[this] var level: Level = null
+  private[this] var useParentsHandlers: Boolean = false
+  private[this] var parent: Logger = null
+  private[this] var filter: Filter = null
+  private[this] var handlers: Array[Handler] = Array.empty
+
+  // Find the effective level
+  private def levelR: Level = {
+    @tailrec
+    def go(logger: Logger): Level = {
+      if (logger.getLevel != null) logger.getLevel
+      else if (logger.getParent == null) null
+      else go(logger.getParent)
+    }
+
+    go(this)
+  }
+
+  // Not implemented, no resource bundle
+  //def getResourceBundle():ResourceBundle = ???
+
+  def getResourceBundleName(): String = resourceBundle
+
+  def setFilter(filter: Filter): Unit = this.filter = filter
+
+  def getFilter(): Filter = filter
+
+  def log(record: LogRecord): Unit = {
+    if (isLoggable(record.getLevel)) {
+      if (useParentsHandlers) {
+        val parent = getParent()
+        if (parent != null) parent.log(record)
```

Comment:
`parent.log(record)` on the next line. We avoid putting statements on the same line of the `if`, we only do it for expressions.